File: datasets/Part 2 - Regression/Section 4 - Simple Linear Regression/1_Regresion_Lineal_Simple.R
# Regresion Lineal Simple


# Importar el dataset
dataset <- read.csv('Salary_Data.csv')


# Dividir los datos en training y testing
install.packages("caTools")
library(caTools)
set.seed(123)
split <- sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
testing_set = subset(dataset, split == FALSE)

# Ajustar el modelo de regresi?n lineal simple con el conjunto de entrenamiento
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)
summary(regressor)

# Predecir resultados con el conjunto de Test
y_pred = predict(regressor, newdata = testing_set)

# Visualizaci?n de los resultados en el conjunto de entrenamiento
#install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, 
                y = predict(regressor, newdata = training_set)),
            colour = "blue") +
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)") +
  xlab("Años de Experiencia") +
  ylab("Sueldo en ($)")

# Visualización de los resultados en el conjunto de testing
#install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = testing_set$YearsExperience, y = testing_set$Salary), 
             colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, 
                y = predict(regressor, newdata = training_set)),
            colour = "blue") +
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Testing)") +
  xlab("Años de Experiencia") +
  ylab("Sueldo en ($)")

File: datasets/Part 2 - Regression/Section 4 - Simple Linear Regression/1_Regresion_Lineal_Simple.py
# -*- coding: utf-8 -*-
"""
Created on Thu Feb  2 08:36:56 2023

@author: Imanol
"""

# Cómo importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


# Importar el data set
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:,:-1].values
y = dataset.iloc[:, 1].values


# Dividir el data set en training y testing
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)


# Crear modelo de Regresión Lineal Simple del conjunto de entrenamiento
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(X_train, y_train)

# Predecir el conjunto de test
y_pred = regression.predict(X_test)

# Visualizar los resultados de entrenamiento
plt.scatter(X_train, y_train, color="red")
plt.plot(X_train, regression.predict(X_train), color="blue")
plt.title("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo (en $)")
plt.show()

# Visualizar los resultados de test
plt.scatter(X_test, y_test, color="red")
plt.plot(X_train, regression.predict(X_train), color="blue")
plt.title("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo (en $)")
plt.show()